File: java/100/LeetCode0102.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;
import util.TreeNode;

public class LeetCode0102 {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> level = new LinkedList<>();
        if (root != null) {
            level.offer(root);
        }
        while (!level.isEmpty()) {
            int size = level.size();
            List<Integer> li = new ArrayList<>();
            for (int i = 0;i < size;++i) {
                TreeNode cur = level.poll();
                li.add(cur.val);
                if (cur.left != null) {
                    level.offer(cur.left);
                }
                if (cur.right != null) {
                    level.offer(cur.right);
                }
            }
            res.add(li);
        }
        return res;
    }
}

File: cpp/L100/LeetCode0102.cc
#include <queue>
#include <vector>
using std::vector;
#include "../heads/tree_node.h"

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        std::queue<TreeNode*> q;
        vector<vector<int>> res;
        if (root == nullptr) {
            return res;
        }
        q.push(root);
        while (!q.empty()) {
            int qz = q.size();
            vector<int> level;
            for (int i = 0;i < qz;++i) {
                TreeNode *node = q.front();
                q.pop();
                level.push_back(node->val);
                if (node->left != nullptr) {
                    q.push(node->left);
                }
                if (node->right != nullptr) {
                    q.push(node->right);
                }
            }
            res.emplace_back(level);
        }
        return res;
    }
};